### PR TITLE
4-rdm from device

### PR DESCRIPTION
**Pull request by erikkjellgren**

### CODE REVIEW

**Comment by ziechys** [comment]
> * Fetching all the RDM elements is expensive, 4-RDM takes ~1000 sec for a (4,4) to build after all the Paulis have been pre-calculated. I suspect that using qiskit's Pauli() class needs to be limited as much as possible, it seems slow. And code that uses generic str instead is easier to optimize.

I agree, the Pauli, PauliList and other internal Qiskit objects are a bit cumbersome either way to keep track of.

**Comment by ziechys** [comment]
> * The code is very memory intensive (estimate missing), the save distributions prob. need to be saved as binary instead of a string of 0 and 1.

So at the moment we just save the cliques as binary string by convenience. We actually explicitly convert to them here:  
``` 
 # Get quasi-distribution in binary probabilities
distr = [res.binary_probabilities() for res in job.result().quasi_dists]
```

This is not needed really. We could save them as number (decimal) and just have a function converting between binary and decimal when needed. E.g. for getting the bitstring sign and stuff like this

File: slowquant/qiskit_interface/linear_response/naive.py [hunk 1]
```diff
@@ -30,7 +30,7 @@
 class quantumLR(quantumLRBaseClass):
     def run(
         self,
-        do_rdm: bool = False,
+        do_rdm: bool = True,
```
Comment (ziechys) [inline slowquant/qiskit_interface/linear_response/naive.py:33]: It is fine for now to have True. 
But I think once we would understand a bit more what the implications of RDMs on shot noise is, we might revisit it :) 
What I also like to add is that by having default True here, means that it is by default a different approach than allproj that cannot use RDMs.

File: slowquant/qiskit_interface/util.py [hunk 2]
```diff
@@ -68,12 +68,14 @@ def get_bitstring_sign(op: Pauli, binary: str) -> int:
...
+    count = 0
     for i, pauli in enumerate(op.to_label()):
         if not pauli == "I":
             if binary[i] == "1":
-                sign = sign * (-1)
-    return sign
+                count += 1
+    if count % 2 == 1:
+        return -1
+    return 1
```
Comment (ziechys) [inline slowquant/qiskit_interface/util.py:78]: Is there a reason for this re-write?

Comment (erikkjellgren) [reply slowquant/qiskit_interface/util.py:78]: The line `sign = sign * (-1)` was a bottleneck.
This is much faster

**Review by ziechys — APPROVED** [review]

**Pull request merged**